lib/utils/file_uploader.dart

```diff
@@ -476,23 +436,142 @@ class FileUploader {
       }
       rethrow;
     } finally {
-      if (mediaUploadData != null && mediaUploadData.sourceFile != null) {
-        // delete the file from app's internal cache if it was copied to app
-        // for upload. Shared Media should only be cleared when the upload
-        // succeeds.
-        if (io.Platform.isIOS ||
-            (uploadCompleted && file.isSharedMediaToAppSandbox())) {
-          await mediaUploadData.sourceFile.delete();
-        }
-      }
-      if (io.File(encryptedFilePath).existsSync()) {
-        await io.File(encryptedFilePath).delete();
-      }
-      if (io.File(encryptedThumbnailPath).existsSync()) {
-        await io.File(encryptedThumbnailPath).delete();
+      await _onUploadDone(
+        mediaUploadData,
+        uploadCompleted,
+        file,
+        encryptedFilePath,
+        encryptedThumbnailPath,
+      );
+    }
+  }
+
+  /*
+  // _mapToExistingUpload links the current file to be uploaded with the
+  // existing files. If the link is successful, it returns true other false.
+   When false, we should go ahead and re-upload or update the file
+    It performs following checks:
+    a) Uploaded file with same localID and destination collection. Delete the
+     fileToUpload entry
+    b) Uploaded file in destination collection but with missing localID.
+     Update the localID for uploadedFile and delete the fileToUpload entry
+    c) A uploaded file exist with same localID but in a different collection.
+    or
+    d) Uploaded file in different collection but missing localID.
+    For both c and d, perform add to collection operation.
+    e) File already exists but different localID. Re-upload
+    In case the existing files already have local identifier, which is
+    different from the {fileToUpload}, then most probably device has
+    duplicate files.
+  */
+  Future<bool> _mapToExistingUploadWithSameHash(
+    MediaUploadData mediaUploadData,
+    File fileToUpload,
+    int toCollectionID,
+  ) async {
+    if (fileToUpload.uploadedFileID != -1 &&
+        fileToUpload.uploadedFileID != null) {
+      _logger.warning('file is already uploaded, skipping mapping logic');
+      return false;
+    }
+    List<String> hash = [mediaUploadData.fileHash];
+    if (fileToUpload.fileType == FileType.livePhoto) {
+      hash.add(mediaUploadData.zipHash);
+    }
+    List<File> existingFiles =
+        await FilesDB.instance.getUploadedFilesWithHashes(
+      hash,
+      fileToUpload.fileType,
+      Configuration.instance.getUserID(),
+    );
+    if (existingFiles?.isEmpty ?? true) {
+      return false;
+    } else {
+      debugPrint("Found some matches");
+    }
+    // case a
+    File sameLocalSameCollection = existingFiles.firstWhere(
+      (element) =>
+          element.uploadedFileID != -1 &&
```

Comment:
When we deserialize from the DB, I think we set it to `null`?

Reply:
Also, if while fetching `existingFiles`, we are already checking for only uploaded files, we could ignore this check?